courses/views.py

```diff
@@ -14,3 +41,43 @@ class CourseRunViewSet(viewsets.ReadOnlyModelViewSet):
     """API for the Program collection"""
     queryset = CourseRun.objects.filter(course__program__live=True)
     serializer_class = CourseRunSerializer
+
+
+class ProgramEnrollmentListView(ListCreateAPIView):
+    """API for the User Program Enrollments"""
+
+    serializer_class = ProgramSerializer
+    permission_classes = (
+        IsAuthenticated,
+    )
+
+    def get_queryset(self):
+        """
+        Filter programs by the user enrollment
+        """
+        queryset = Program.objects.filter(programenrollment__user=self.request.user)
+        return queryset.order_by('title')
+
+    @transaction.atomic
+    def create(self, request, *args, **kwargs):  # pylint: disable=unused-argument
+        """
+        Create an enrollment for the current user
+        """
+        program_id = request.data.get('program_id')
+        if not isinstance(program_id, int):
+            raise ValidationError('A `program_id` parameter must be specified')
+
+        if ProgramEnrollment.objects.filter(user=request.user, program__pk=program_id).exists():
+            raise ResourceConflict('The enrollment for the specified program already exists')
+
+        program_queryset = Program.objects.filter(live=True, pk=program_id)
```

Comment:
Since there can only be one program with a given `pk`, why not use `Program.objects.get` within a try/except or `get_object_or_404`?